src/Rules/DeadCode/UnusedPrivateConstantRule.php

```diff
@@ -59,10 +59,14 @@ public function processNode(Node $node, Scope $scope): array
 
 		foreach ($node->getFetches() as $fetch) {
 			$fetchNode = $fetch->getNode();
-			if (!$fetchNode->class instanceof Node\Name) {
+			if (!$fetchNode->name instanceof Node\Identifier) {
 				continue;
 			}
-			if (!$fetchNode->name instanceof Node\Identifier) {
+			if ($fetchNode->class instanceof Node\Expr\Variable && $fetchNode->class->name === 'this') {
```

Comment:
This isn't a general fix for this problem. For example it's not gonna fix this scenario:

```php
$self = $this;
echo $self::HELLO;
```

or this:

```php
$self = new self();
echo $self::HELLO;
```

The correct fix this: Acknowledge that ClassConstFetch::$class can contain either a Name or an Expr.

If it's a Name then just check the class name using `$scope->resolveName()`. If it's an Expr, get type using `$scope->getType()`. If it's a TypeWithClassName, continue and treat it the same way as the Name in `ClassConstFetch::$class`.

Reply:
Thank you for your great pointers.
I didn't even know that the first  example is a valid syntax! Fixing now.